src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -585,6 +589,112 @@ class spell_pri_penance : public SpellScript
+{
+    PrepareAuraScript(spell_pri_penance_channeled);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PRIEST_POWER_OF_THE_DARK_SIDE });
+    }
+
+    void HandleOnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if (caster->HasAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE))
+                caster->RemoveAura(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectRemove += AuraEffectRemoveFn(spell_pri_penance_channeled::HandleOnRemove, EFFECT_0, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 47666 - Penance (Damage), 47750 - Penance (Healing)
+class spell_pri_penance_damage_healing: public SpellScript
+{
+    PrepareSpellScript(spell_pri_penance_damage_healing);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_PRIEST_POWER_OF_THE_DARK_SIDE });
+    }
+
+    void HandleLaunchTarget(SpellEffIndex effIndex)
+    {
+        if (Unit* caster = GetCaster())
+        {
+            if (AuraEffect* powerOfTheDarkSide = caster->GetAuraEffect(SPELL_PRIEST_POWER_OF_THE_DARK_SIDE, EFFECT_0))
+            {
+                PreventHitDefaultEffect(effIndex);
+
+                uint32 value = 0;
+
+                if (GetSpellInfo()->Id == SPELL_PRIEST_PENANCE_DAMAGE)
+                {
+                    uint32 damageBonus = caster->SpellDamageBonusDone(GetHitUnit(), GetSpellInfo(), GetEffectValue(), SPELL_DIRECT_DAMAGE, GetEffectInfo(effIndex));
+                    value = damageBonus + uint32(damageBonus * GetEffectVariance());
+                    value *= 1.0f + (powerOfTheDarkSide->GetAmount() / 100.0f);
+                    value = GetHitUnit()->SpellDamageBonusTaken(caster, GetSpellInfo(), value, SPELL_DIRECT_DAMAGE);
+                    SetHitDamage(value);
+                }
+                else
+                {
+                    uint32 healingBonus = caster->SpellHealingBonusDone(GetHitUnit(), GetSpellInfo(), GetEffectValue(), HEAL, GetEffectInfo(effIndex));
+                    value = healingBonus + uint32(healingBonus * GetEffectVariance());
+                    value *= 1.0f + (powerOfTheDarkSide->GetAmount() / 100.0f);
+                    value = GetHitUnit()->SpellHealingBonusTaken(caster, GetSpellInfo(), value, HEAL);
+                    SetHitHeal(value);
+                }
+            }
+        }
+    }
+
+    void Register() override
+    {
+        if (m_scriptSpellId == SPELL_PRIEST_PENANCE_DAMAGE)
+            OnEffectLaunchTarget += SpellEffectFn(spell_pri_penance_damage_healing::HandleLaunchTarget, EFFECT_0, SPELL_EFFECT_SCHOOL_DAMAGE);
+        if (m_scriptSpellId == SPELL_PRIEST_PENANCE_HEALING)
+            OnEffectLaunchTarget += SpellEffectFn(spell_pri_penance_damage_healing::HandleLaunchTarget, EFFECT_0, SPELL_EFFECT_HEAL);
```

Comment:
Why exactly are you using OnEffectLaunchTarget and not just OnEffectHitTarget and just add the aura on top of the already existing calculation?